Lib/types.py

```diff
@@ -143,6 +143,34 @@ def _calculate_meta(meta, bases):
                         "of the metaclasses of all its bases")
     return winner
 
+
+def get_original_bases(cls, /):
+    """Return the class's "original" bases prior to modification by `__mro_entries__`.
+
+    This is useful for introspecting Generics.
+
+    Examples::
+
+        from typing import TypeVar, Generic
+
+        T = TypeVar("T")
+        class Foo(Generic[T]): ...
+        class Bar(Foo[int], float): ...
+        class Baz(list[str]): ...
+
+        get_original_bases(Foo) == (Generic[T],)
+        get_original_bases(Bar) == (Foo[int], float)
+        get_original_bases(Baz) == (list[str],)
+        get_original_bases(int) == None
+    """
+    if isinstance(cls, type):
+        try:
+            return cls.__orig_bases__
+        except AttributeError:
+            pass
+    return None
```

Comment:
So, I thought this over yesterday, and I'm not sure we should return `None` from this function if the class passed in doesn't have an `__orig_bases__` attribute. The docstring and docs we've written for this function promise that the function will "return the objects in the bases list of the class's definition as they existed prior to any modification by `__mro_entries__`". But the "original bases list" of `int` isn't `None` -- it's `(<class 'object'>,)`.

Ultimately it's quite hard to explain to users of this function why it might return `None` sometimes. We can't say "well, it returns `None` if the class doesn't have an `__orig_bases__` attribute, because the _whole point_ of this function is so that users don't need to know anything about the `__orig_bases__` attribute.

Maybe we should instead fall back to the `__bases__` attribute if the object doesn't have an `__orig_bases__` attribute -- something like this?

```suggestion
    try:
        return cls.__orig_bases__
    except AttributeError:
        try:
            return cls.__bases__
        except AttributeError:
            raise TypeError(
                f'Expected an instance of type, not {type(cls).__name__!r}'
            ) from None
```

Reply:
I do think this is a better idea than the original my only question would be about this now now not including object as the last entry which strikes me as slightly inconsistent (even if it's technically fine according to the documentation). Should we potentially remove object from `__bases__` or add it to the `__orig_bases__`?

Reply:
I don't see any reason to remove `object` from `__bases__`. `__bases__` gives you the _direct_ bases of the class, not the full `mro`. So `object` is only included in the `__bases__` tuple of a class if the class directly inherits from `object` (which `int` does):

```pycon
>>> class Foo: ...
...
>>> Foo.__bases__
(<class 'object'>,)
>>> class Bar(Foo): ...
...
>>> Bar.__bases__
(<class '__main__.Foo'>,)
```

Reply:
Ah, I see yeah that makes sense then